share/shwap/p2p/shrex/shrexeds/client.go

```diff
@@ -80,35 +86,40 @@ func (c *Client) RequestEDS(
 
 func (c *Client) doRequest(
 	ctx context.Context,
-	dataHash share.DataHash,
+	root *share.AxisRoots,
+	height uint64,
 	to peer.ID,
 ) (*rsmt2d.ExtendedDataSquare, error) {
 	streamOpenCtx, cancel := context.WithTimeout(ctx, c.params.ServerReadTimeout)
 	defer cancel()
 	stream, err := c.host.NewStream(streamOpenCtx, to, c.protocolID)
 	if err != nil {
-		return nil, fmt.Errorf("failed to open stream: %w", err)
+		return nil, fmt.Errorf("open stream: %w", err)
 	}
 	defer stream.Close()
 
 	c.setStreamDeadlines(ctx, stream)
-
-	req := &pb.EDSRequest{Hash: dataHash}
-
 	// request ODS
-	log.Debugw("client: requesting ods", "hash", dataHash.String(), "peer", to.String())
-	_, err = serde.Write(stream, req)
+	log.Debugw("client: requesting ods",
+		"height", height,
+		"peer", to.String())
+	id, err := shwap.NewEdsID(height)
+	if err != nil {
+		return nil, fmt.Errorf("create request: %w", err)
```

Comment:
It also needs a stream reset. We need to check that all the error reset it, otherwise we may leak them